src/dolphin/downloadDolphin.ts

```diff
@@ -145,100 +146,97 @@ async function installDolphin(
     }
     case "linux": {
-      // Delete the existing app image if there is one
-      try {
-        const dolphinAppImagePath = await findDolphinExecutable(type);
-        // Delete the existing app image if it already exists
-        if (await fileExists(dolphinAppImagePath)) {
-          log(`${dolphinAppImagePath} already exists. Deleting...`);
-          await fs.remove(dolphinAppImagePath);
-        }
-      } catch (err) {
-        log("No existing AppImage found");
-      }
-
-      if (cleanInstall) {
-        const userFolder = type === DolphinLaunchType.NETPLAY ? "SlippiOnline" : "SlippiPlayback";
-        await fs.remove(path.join(app.getPath("home"), ".config", userFolder));
-      }
-
-      const zip = new AdmZip(assetPath);
-      zip.extractAllTo(dolphinPath, true);
-
-      // Make the file executable
-      const dolphinAppImagePath = await findDolphinExecutable(type);
-      log(`Setting executable permissions...`);
-      await fs.chmod(dolphinAppImagePath, "755");
+      await installDolphinOnLinux(type, assetPath, dolphinPath, log);
       break;
     }
     default: {
       throw new Error(`Installing Netplay is not supported on this platform: ${process.platform}`);
     }
   }
 }
+
+async function installDolphinOnMac(assetPath: string, dolphinPath: string, log: (message: string) => void) {
+  const backupLocation = dolphinPath + "_old";
+  const dolphinResourcesPath = path.join(dolphinPath, "Slippi Dolphin.app", "Contents", "Resources");
+
+  const alreadyInstalled = await fs.pathExists(dolphinResourcesPath);
+  let shouldRestore = false;
+  if (alreadyInstalled) {
+    log(`${dolphinResourcesPath} already exists. Moving...`);
+    await fs.move(dolphinPath, backupLocation);
+    shouldRestore = true;
+  }
+
+  log(`Extracting to: ${dolphinPath}`);
+  await extractDmg(assetPath, dolphinPath);
+  const files = await fs.readdir(dolphinPath);
+  await Promise.all(
+    files
+      .filter((file) => file !== "Slippi Dolphin.app")
+      .map(async (file) => {
+        await fs.remove(path.join(dolphinPath, file));
+      }),
+  );
+
+  // sometimes permissions aren't set properly after the extraction so we will forcibly set them on install
+  const binaryLocation = path.join(dolphinPath, "Slippi Dolphin.app", "Contents", "MacOS", "Slippi Dolphin");
+  const userInfo = os.userInfo();
+  await fs.chmod(path.join(dolphinPath, "Slippi Dolphin.app"), "777");
+  await fs.chown(path.join(dolphinPath, "Slippi Dolphin.app"), userInfo.uid, userInfo.gid);
+  await fs.chmod(binaryLocation, "777");
+  await fs.chown(binaryLocation, userInfo.uid, userInfo.gid);
+
+  // move backed up User folder and user.json
+  if (alreadyInstalled) {
+    const oldUserFolder = path.join(backupLocation, "Slippi Dolphin.app", "Contents", "Resources", "User");
+    const newUserFolder = path.join(dolphinResourcesPath, "User");
+
+    if (shouldRestore) {
+      log("moving User folder...");
+      await fs.move(oldUserFolder, newUserFolder);
+    }
```

Comment:
Likewise if `alreadyInstalled` is true, then `shouldRestore` would also be guaranteed to be true. Since the only alternative is that `await fs.move` errors in which case we would never get to the rest of the function anyways. I think we can just completely remove the `shouldRestore` variable.